src/Evolution/Systems/Ccz4/TimeDerivative.cpp

```diff
@@ -0,0 +1,625 @@
+      inv_spatial_metric, (*conformal_factor_squared)() *
+                              (*inv_conformal_spatial_metric)(ti_I, ti_J));
+
+  ::TensorExpressions::evaluate<ti_I, ti_J>(
+      inv_a_tilde, a_tilde(ti_k, ti_l) *
+                       (*inv_conformal_spatial_metric)(ti_I, ti_K) *
+                       (*inv_conformal_spatial_metric)(ti_J, ti_L));
+
+  for (size_t i = 0; i < num_points; i++) {
+    get(*lapse)[i] = exp(get(ln_lapse)[i]);
+  }
+
+  if (slicing_condition_type == SlicingConditionType::Harmonic) {
+    get(*slicing_condition) = 1.0;
+    get(*d_slicing_condition) = 0.0;
+    get(*lapse_times_slicing_condition) = get(*lapse);
+  } else if (slicing_condition_type == SlicingConditionType::Log) {
+    ASSERT(min(get(*lapse)) > 0.0,
+           "The lapse must be positive when using "
+           "Ccz4::SlicingConditionType::Harmonic.");
+
+    // g(\alpha) == 2 / \alpha
+    get(*slicing_condition) = 2.0 / get(*lapse);
+    // g'(\alpha)  == -2 / \alpha^2 == -0.5 * g(\alpha)^2
+    get(*d_slicing_condition) = -0.5 * square(get(*slicing_condition));
+    // \alpha g(\alpha)  == 2
+    get(*lapse_times_slicing_condition) = 2.0;
+  } else {
+    ERROR("Unknown Ccz4::SlicingConditionType");
+  }
+
+  // expressions and identities needed for evolution equations: eq 13 - 27
+
+  // eq 13
+  ::TensorExpressions::evaluate(
+      trace_a_tilde,
+      (*inv_conformal_spatial_metric)(ti_I, ti_J) * a_tilde(ti_i, ti_j));
+
+  // eq 14
+  ::TensorExpressions::evaluate<ti_k, ti_I, ti_J>(
+      field_d_up, (*inv_conformal_spatial_metric)(ti_I, ti_N) *
+                      (*inv_conformal_spatial_metric)(ti_M, ti_J) *
+                      field_d(ti_k, ti_n, ti_m));
+
+  // eq 15
+  ::Ccz4::conformal_christoffel_second_kind(conformal_christoffel_second_kind,
+                                            *inv_conformal_spatial_metric,
+                                            field_d);
+
+  // eq 16
+  ::Ccz4::deriv_conformal_christoffel_second_kind(
+      d_conformal_christoffel_second_kind, *inv_conformal_spatial_metric,
+      field_d, d_field_d, *field_d_up);
+
+  // eq 17
+  ::Ccz4::christoffel_second_kind(christoffel_second_kind,
+                                  conformal_spatial_metric,
+                                  *inv_conformal_spatial_metric, field_p,
+                                  *conformal_christoffel_second_kind);
+
+  // temporary expressions needed for eq 18 - 20
+  ::TensorExpressions::evaluate<ti_l>(
+      contracted_christoffel_second_kind,
+      (*christoffel_second_kind)(ti_M, ti_l, ti_m));
+
+  ::TensorExpressions::evaluate<ti_i, ti_j>(
+      contracted_d_conformal_christoffel_difference,
+      (*d_conformal_christoffel_second_kind)(ti_m, ti_M, ti_i, ti_j) -
+          (*d_conformal_christoffel_second_kind)(ti_j, ti_M, ti_i, ti_m));
+
+  ::TensorExpressions::evaluate<ti_L>(contracted_field_d_up,
+                                      (*field_d_up)(ti_m, ti_M, ti_L));
```

Comment:
Note that as part of rebasing, I added these above three temporary expression computations for the Ricci tensor (eq 18 - 20) because `Ccz4::spatial_ricci_tensor`'s signature was recently updated to take these temporaries.